### ISSUE TITLE
resource/aws_rds_cluster_instance is being destroyed and re-created when updating engine_version while apply_immediately is set to false

### ISSUE THREAD

**Issue by tsiq-vincentc** [hashicorp/terraform-provider-aws#10714]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v.0.12.8
provider.aws v2.34.0

### Affected Resource(s)

* aws_rds_cluster_instance

### Terraform Configuration Files

*Initial applied config*

```hcl
resource "aws_rds_cluster_instance" "main" {
  count                        = 2
  identifier                   = "test-tf-version-${count.index}"
  cluster_identifier           = aws_rds_cluster.main.id
  instance_class               = "db.r5.large"
  publicly_accessible          = false
  engine                       = "aurora-postgresql"
  engine_version               = "10.6"
  apply_immediately            = false
  auto_minor_version_upgrade   = false
}

resource "aws_rds_cluster" "main" {
  cluster_identifier              = "test-tf-version"
  master_username = "postgresql"
  master_password = "postgresql"
  skip_final_snapshot             = false
  apply_immediately               = false
  engine                          = "aurora-postgresql"
  engine_version                  = "10.6"
  storage_encrypted               = true
  deletion_protection             = true
}
```

*Config to be applied* 

```hcl
resource "aws_rds_cluster_instance" "main" {
  count                        = 2
  identifier                   = "test-tf-version-${count.index}"
  cluster_identifier           = aws_rds_cluster.main.id
  instance_class               = "db.r5.large"
  publicly_accessible          = false
  engine                       = "aurora-postgresql"
  engine_version               = "10.7"
  apply_immediately            = false
  auto_minor_version_upgrade   = false
}

resource "aws_rds_cluster" "main" {
  cluster_identifier              = "test-tf-version"
  master_username = "postgresql"
  master_password = "postgresql"
  skip_final_snapshot             = false
  apply_immediately               = false
  engine                          = "aurora-postgresql"
  engine_version                  = "10.7"
  storage_encrypted               = true
  deletion_protection             = true
}
```

### Expected Behavior

As `apply_immediately` is set to false, when applying the config with a version bump, there should be only `update in-place` actions for the both the cluster and the instances. This should put a pending modification for the cluster in AWS RDS, where the version update will be done during the next maintenance window.

### Actual Behavior

Plan output:
```
Terraform will perform the following actions:

  # aws_rds_cluster.main will be updated in-place
  ~ resource "aws_rds_cluster" "main" {
        apply_immediately                   = false
        arn                                 = "redacted"
        availability_zones                  = [
            "us-east-1a",
            "us-east-1b",
            "us-east-1c",
        ]
        backtrack_window                    = 0
        backup_retention_period             = 1
        cluster_identifier                  = "test-tf-version"
        cluster_members                     = [
            "test-tf-version-0",
            "test-tf-version-1",
        ]
        cluster_resource_id                 = "redacted"
        copy_tags_to_snapshot               = false
        db_cluster_parameter_group_name     = "default.aurora-postgresql10"
        db_subnet_group_name                = "default"
        deletion_protection                 = true
        enabled_cloudwatch_logs_exports     = []
        endpoint                            = "redacted"
        engine                              = "aurora-postgresql"
        engine_mode                         = "provisioned"
      ~ engine_version                      = "10.6" -> "10.7"
        hosted_zone_id                      = "redacted"
        iam_database_authentication_enabled = false
        iam_roles                           = []
        id                                  = "test-tf-version"
        kms_key_id                          = "redacted"
        master_password                     = (sensitive value)
        master_username                     = "postgresql"
        port                                = 5432
        preferred_backup_window             = "05:56-06:26"
        preferred_maintenance_window        = "tue:03:38-tue:04:08"
        reader_endpoint                     = "redacted"
        skip_final_snapshot                 = false
        storage_encrypted                   = true
        tags                                = {}
        vpc_security_group_ids              = [
            "sg-05861584e0fa01828",
        ]
    }

  # aws_rds_cluster_instance.main[0] must be replaced
-/+ resource "aws_rds_cluster_instance" "main" {
        apply_immediately               = false
      ~ arn                             = "redacted" -> (known after apply)
        auto_minor_version_upgrade      = false
      ~ availability_zone               = "us-east-1b" -> (known after apply)
        cluster_identifier              = "test-tf-version"
        copy_tags_to_snapshot           = false
      ~ db_parameter_group_name         = "default.aurora-postgresql10" -> (known after apply)
      ~ db_subnet_group_name            = "default" -> (known after apply)
      ~ dbi_resource_id                 = "redacted" -> (known after apply)
      ~ endpoint                        = "redacted" -> (known after apply)
        engine                          = "aurora-postgresql"
      ~ engine_version                  = "10.6" -> "10.7" # forces replacement
      ~ id                              = "test-tf-version-0" -> (known after apply)
        identifier                      = "test-tf-version-0"
      + identifier_prefix               = (known after apply)
        instance_class                  = "db.r5.large"
      ~ kms_key_id                      = "redacted" -> (known after apply)
        monitoring_interval             = 0
      + monitoring_role_arn             = (known after apply)
      ~ performance_insights_enabled    = false -> (known after apply)
      + performance_insights_kms_key_id = (known after apply)
      ~ port                            = 5432 -> (known after apply)
      ~ preferred_backup_window         = "05:56-06:26" -> (known after apply)
      ~ preferred_maintenance_window    = "thu:07:07-thu:07:37" -> (known after apply)
        promotion_tier                  = 0
        publicly_accessible             = false
      ~ storage_encrypted               = true -> (known after apply)
      - tags                            = {} -> null
      ~ writer                          = true -> (known after apply)
    }

  # aws_rds_cluster_instance.main[1] must be replaced
-/+ resource "aws_rds_cluster_instance" "main" {
        apply_immediately               = false
      ~ arn                             = "redacted" -> (known after apply)
        auto_minor_version_upgrade      = false
      ~ availability_zone               = "us-east-1c" -> (known after apply)
        cluster_identifier              = "test-tf-version"
        copy_tags_to_snapshot           = false
      ~ db_parameter_group_name         = "default.aurora-postgresql10" -> (known after apply)
      ~ db_subnet_group_name            = "default" -> (known after apply)
      ~ dbi_resource_id                 = "redacted" -> (known after apply)
      ~ endpoint                        = "redacted" -> (known after apply)
        engine                          = "aurora-postgresql"
      ~ engine_version                  = "10.6" -> "10.7" # forces replacement
      ~ id                              = "test-tf-version-1" -> (known after apply)
        identifier                      = "test-tf-version-1"
      + identifier_prefix               = (known after apply)
        instance_class                  = "db.r5.large"
      ~ kms_key_id                      = "redacted" -> (known after apply)
        monitoring_interval             = 0
      + monitoring_role_arn             = (known after apply)
      ~ performance_insights_enabled    = false -> (known after apply)
      + performance_insights_kms_key_id = (known after apply)
      ~ port                            = 5432 -> (known after apply)
      ~ preferred_backup_window         = "05:56-06:26" -> (known after apply)
      ~ preferred_maintenance_window    = "sat:09:56-sat:10:26" -> (known after apply)
        promotion_tier                  = 0
        publicly_accessible             = false
      ~ storage_encrypted               = true -> (known after apply)
      - tags                            = {} -> null
      ~ writer                          = false -> (known after apply)
    }

Plan: 2 to add, 1 to change, 2 to destroy.
```

The action on the cluster is as expected, however the instances are being destroyed and re-created in the plan.

### Steps to Reproduce

1. `terraform apply` the initial config
2. Update config to the second config listed.
3. `terraform plan`.

**Comment by chaspy** [comment]
`engine_version` requires force_new re-create 😞 
https://github.com/terraform-providers/terraform-provider-aws/blob/v2.34.0/aws/resource_aws_rds_cluster_instance.go#L104

**Comment by mccartney** [comment]
@justinretzolk your comment seems to have a typo - `3.36.0` vs `3.63.0`.

(Also FYI - this issue has been mentioned in https://stackoverflow.com/a/69846183/118587)

**Comment by justinretzolk** [comment]
Thanks for the heads up @mccartney! I've updated the comment so as to not cause any further confusion.

**Comment by justinretzolk** [comment]
Since this issue has been resolved by the `3.63.0` releases (thanks again for letting me know about the previous typo!) I'm going to close this issue. If you feel I've done this in error, please do let us know.
